src/pages/LiveChat/components/messages/Message.tsx

```diff
@@ -12,20 +12,25 @@ type Props = {
 export default function Message({ message, messagesListDb, messagesListActionRef }: Props) {
   const { user } = useOutletContext() as LiveChatContext;
 
-  let liClassName = "mb-1 mx-3";
-  let pClassName = "block w-fit max-w-[80%] bg-gradient-to-br from-secondary to-primary px-3 py-1 rounded-2xl";
-  if (user?.uid === message.userId) {
-    pClassName += " ml-auto";
-  }
+  /*SG using defined variables for combining tailwind classes is more of a antipattern in tailwind if is used in one place
+    is better to create a custom utility class if it's a style applied on multiple elements
+    tailwind has utility classes meant to avoid using names for variables
+  */
 
   const handleDeleteMessage = () => {
-    deleteDoc(doc(messagesListDb, message.id));
-    messagesListActionRef.current.action = "deleted";
+    // SG As far as I can see deleteDoc is returning a promise, you can use .then to apply the action to the ref after the promise is fulfilled
```

Comment:
I'm not using .then() here because .then() will execute the code only when the response from the server is received which may be after some x ms, 1s, or etc depending on the network speed. I want to auto-scroll(or not) in real time with no delay depending on the user's action even if the call is executed with some delay, or not at all. Chat feels more responsive this way.